kotlin-inject-compiler-core/src/main/kotlin/me/tatarka/inject/compiler/InjectGenerator.kt

```diff
@@ -628,13 +628,28 @@ fun AstElement.isComponent() = hasAnnotation<Component>()
 
 fun AstMethod.isProvides(): Boolean = hasAnnotation<Provides>()
 
-fun AstClass.isInject(options: Options): Boolean {
-    if (options.enableJavaxAnnotations) {
-        if (primaryConstructor?.hasAnnotation("javax.inject.Inject") == true) {
-            return true
+fun AstClass.findInjectConstructors(messenger: Messenger, options: Options): AstConstructor? {
+    val injectCtors = constructors?.filter {
+        if (options.enableJavaxAnnotations) {
+            it.hasAnnotation("javax.inject.Inject") || it.hasAnnotation<Inject>()
+        } else {
+            it.hasAnnotation<Inject>()
+        }
+    }.orEmpty()
+
+    return when {
+        hasAnnotation<Inject>() && injectCtors.isNotEmpty() -> {
+            messenger.error("Annotating constructors with @Inject in @Inject-annotated classes is not supported.", this)
```

Comment:
Would like to match the wording of other error messages so something like:
```suggestion
            messenger.error("Cannot annotate constructor with @Inject in an @Inject-annoated class", this)
```